datareduction/datareductionoperator.cpp

```diff
@@ -25,6 +25,8 @@
 #include <iostream>
 #include <limits>
 #include <array>
+#include <memory>
```

Comment:
Let's not include development benchmarking stuff in the actual PR (Also for the clock() calls and printfs below)